feat: add simple linking to the examples (#702)
substrait-io/substrait: site/docs/tutorial/examples.md

# Code samples and examples

> It's very useful to have examples of how APIs are used; both to get information on teh best practicses for using APIs and ideas of how they can be used. 

Reviewer: teh

Author: thanks @jacques-n   when the PR in the substrait-java is merged.. I'll update this one

> Each language binding is intended to contain examples that are relavent to that language. Hoped that this list will grow over time, new submissions always welcome.

Reviewer 2: ```suggestion
Each language binding is intended to contain examples that are relevant to that language. New contributions are always welcome.
```

## Java

> - [Substrait-Spark](https://github.com/substrait-io/substrait-java/tree/main/examples/substrait-spark) this demonstrates how Substrait plans can be created and consumed within Apache Spark. The examples run within a simple Spark cluster of a couple of docker containers.

Reviewer 2: ```suggestion
- [Substrait-Spark](https://github.com/substrait-io/substrait-java/tree/main/examples/substrait-spark) this demonstrates how Substrait plans can be created and consumed within Apache Spark. The examples run within a simple Spark cluster, composed by a couple of Docker containers.
```
